provider/fs/src/export/fs_exporter.rs

```diff
@@ -163,6 +167,16 @@ impl DataExporter for FilesystemExporter {
         Ok(())
     }
 
+    fn flush_with_fallback(&self, key: DataKey, _: FallbackMode) -> Result<(), DataError> {
+        let mut path_buf = self.root.clone().into_os_string();
+        write!(&mut path_buf, "/{key}").expect("infallible");
+
+        fs::create_dir_all(&path_buf)
+            .map_err(|e| DataError::from(e).with_path_context(&path_buf))?;
```

Comment:
Suggestion: it might be useful to add a file like `.gitkeep` to the directory to prevent VCS systems from removing the empty directory.